system/inc/system_cloud.h

```diff
@@ -240,6 +240,23 @@ void spark_cloud_flag_disconnect(void);    // should be set connected since it m
  */
 bool spark_cloud_flag_auto_connect(void);
 
+/**
+ * Option flags changing the behavior of `spark_cloud_disconnect()`.
+ */
+typedef enum cloud_disconnect_flag {
+    CLOUD_DISCONNECT_GRACEFULLY = 0x01, ///< Disconnect gracefully.
+    CLOUD_DISCONNECT_DONT_CLOSE = 0x02 ///< Do not close the socket.
```

Comment:
What does it mean to disconnect without closing the socket?

Reply:
It doesn't make much sense from the user's perspective, but we use it internally as an optimization. These flags are also defined in [#1899](https://github.com/particle-iot/device-os/pull/1899/files#diff-b43faa0ec0ffb23fa014851243cadf77R87), so having the same set of flags in this PR will simplify merging.